Chess/src/chess/Cell.java

```diff
@@ -132,4 +134,39 @@ public boolean ischeck() // Function to check if the current cell is in check
     {
         return ischeck;
     }
+    
+    @Override
+    public boolean equals(Object o) {

```

Comment:
This method is not necessary because we don't really ever call this, BUT I see some merit it having it. If you verify that `this.x == that.y && this.y == that.y` first, you can simplify this logic quite a bit.

Reply:
but just x == x and y == y isnt enough to determine if they are the same anyways so it doesnt matter if you verify those first or not

Reply:
If you know that x and y are the same, you can then simply return false if the `getpiece()` values are not equal, please correct me if I'm wrong

Reply:
```java
if (this.getpiece() == null) {
    if (that.getpiece() != null) return false;
    return true;
}
return this.getpiece().equals(that.getpiece());

Reply:
the null values would still be needed to be checked because I can't call get piece if its null. the amount of code would change by 1 line which I dont think is a significant change.

Reply:
The null values are being checked here, could you please give me as configuration where it would break
```java
if (this == o) return true;
if (!(o instanceof Cell)) return false;
Cell that = (Cell) o;
if (this.getpiece() == null) {
    if (that.getpiece() != null) return false;
    return true;
}
return this.getpiece().getPieceType().equals(that.getpiece().getPieceType());